commands/files/file.go

```diff
@@ -11,6 +11,57 @@ var (
 	ErrNotReader    = errors.New("This file is a directory, can't use Reader functions")
 )
 
+// An AdvReader is like a Reader but supports getting the current file
+// path and offset into the file when applicable.
+type AdvReader interface {
+	io.Reader
+	ExtraInfo() ExtraInfo
```

Comment:
hrm... I'll look at this in more detail once we get the blocks changes in and this PR rebased on top of that.